common-libs/internal-api-clients/src/services/GenericApiService.ts

```diff
@@ -71,8 +71,8 @@ export default class GenericApiService<TApi extends BuiltApiType> {
 
     // eslint-disable-next-line no-unused-vars
     const path = pathTemplate.replace(/\{(\w+)\}/g, (_match, p1) => options.pathParams?.[p1])
-    const url = new URL(path)
-    url.search = new URLSearchParams(options.queryParams ?? {}).toString()
+    const search = new URLSearchParams(options.queryParams ?? {}).toString()
+    const url = new URL(path.endsWith('?') || !search ? path + search : `${path}?${search}`)
```

Comment:
It was not a mutation, it was using URL API to build a new URL instead of manually composing URL string.
AFAIK the setters are there in https://url.spec.whatwg.org/ specifically to allow developers to build URLs from parts instead of manually composing strings.

Are there any other ways to solve this? Maybe `expo` provides some other similar component to build URLs, if its own URL object does not implement the standard?

Reply:
Also do we have any link describing incompatibility between expo's URL and whatwg's standard URL?

Reply:
maybe we can use a native JS library to compose the URL from parts but IMHO no need to add a new package, or change strategies according to platform just for avoiding string concatenation

Reply:
What was the problem here?

Reply:
How about this?
```js
const url = new URL(path)
for (const [key, value] of Object.entries(options.queryParams)) {
    url.searchParams.set(key, value)
}
```

Reply:
@Yengas , an answer to that would probably help us in the future:

> Also do we have any link describing incompatibility between expo's URL and whatwg's standard URL?